### DISCUSSION TITLE
Apply native filters to annotation layers

### DISCUSSION THREAD

**Discussion by Gradsta** [apache/superset#22836] [Ideas]
Dashboard native filters don't appear to be pushed down to the datasets that drive annotation layers. 

I recently tried to visualize an upper bound using a time series annotation layer on a line chart. I wanted to use an annotation layer because I wanted to control the style of the upper bound line. Unfortunately, my annotation layer did not update when I applied a filter from my dashboard.

This may be related to #18744.

**Comment by mdelarosa17** [comment]
+1 this is definitely something that would make annotations a game-changer.

**Comment by zhxgigi** [comment]
Our requirement is to have dynamic annotation based on filtering conditions in a line chart, and currently it seems that the annotation layer is not able to respond to filtering conditions.

**Reply by mdelarosa17** [reply]
Found a workaround - annoying but it works. For our needs, we need to flag releases by region but default to US. 

1. Create a jinja dataset with your annotations
2. Update the query in the jinja to use the filters from the dashboard.
3. Attach annotations to chart

So it'll work by default for the US but applying a different region filter appears to keep annotation for the US. However, refreshing the dashboard after the filter has changed applies the correct annotation.

Its not a great user experience but it does work.

**Comment by trungvq53** [comment]
This! Together with the ability to change orientation, font and size of annotation text.

**Comment by KMCzajkowski** [comment]
Still interested in that in 2026. Has any progress been made on this?
